src/main/java/cat/udl/eps/entsoftarch/textannot/domain/TagHierarchy.java

```diff
@@ -23,10 +18,14 @@
     private Integer tagHierachyId;
 
     @NotBlank
+    @Column(unique = true)
     private String name;
 
     @Override
     public Integer getId() {
         return this.tagHierachyId;
     }
+
+    @OneToMany(mappedBy = "definedIn")
```

Comment:
TODO: remove to avoid bidirectional relationships as agreed. Use findBy... repository method instead